doc/architectural/compilation-and-development.md

```diff
@@ -109,42 +109,68 @@ So the first command will run all the CORE tests in `regression/cbmc/cpp` and
 list which tests it will run without actually running them.
 
 
-\subsubsection compilation-and-development-subsubsection-running-regression-tests-directly-with-test-pl Running regression tests directly with `test.pl`
+\subsubsection compilation-and-development-subsubsection-running-individual-regression-tests-directly-with-test-pl Running individual regression tests directly with test.pl
 
 It can be useful to run a single test folder in isolation. This can be done by
 running `regression/test.pl` directly. The way that `test.pl` is run varies between the
 different modules, and can be ascertained by looking at the `test` target in
 the makefile. The simple case is when there isn't a file called `chain.sh`.
 Then you can directly run `test.pl` on a single test folder with the
-following command (from the module directory):
+following command from the module directory (note that it is recommended to
+use absolute paths as it avoids many issues):
 
-    ../test.pl -p -c <absolute-path-to-binary> <test-folder>
+    <absolute-path-to-test.pl> -p -c <absolute-path-to-binary> <test-folder>
 
 `-p` makes it print a log of failed tests and `-c` tells it where to find the
-binary to run (the path does not have to be absolute, but it is recommended).
+binary to run, e.g. `cbmc`, `jbmc` or `goto-analyzer`.
 If `<test-folder>` is not provided then all test directories are run. The
 `--help` option lists all command line options, including `-j` for running
 multiple tests in parallel and `-C`, `-T`, `-F` and `-K` for controlling
 whether `CORE`, `THOROUGH`, `FUTURE` or `KNOWNBUG` tests are run.
 
-When there is a file called `chain.sh` then `test.pl` is called with
-`-c ../chain.sh`, followed by arguments which vary from module to module.
+When there is a file called `chain.sh` then the command should look like
+
+    <absolute-path-to-test.pl> -p -c '<absolute-path-to-chain-sh> <arg-1> ... <arg-n>' <test-folder>
+
+Note the single quotes - it will not work without them. What to put for the
```

Comment:
I'd phrase that something like "the test binary and its initial command line arguments are _single_ value for the `-c` option, so must be be wrapped in quotes if there are any unescaped spaces in there"